### DISCUSSION TITLE
Java Publisher, C++ Subscriber

### DISCUSSION THREAD

**Discussion by kswain1906** [OpenDDS/OpenDDS#4099] [Q&A]
I have a Java publisher and C++ subscriber, both using the same domain ID and topic, but published messages are not being received by the subcriber.  I was able to send and receive with Java publisher to Java subscriber, as well as C++ publisher to C++ subscriber.  Are there known issues with exchanging messages between C++ and Java apps? Are there any recommendations for monitoring the registered domain participants to ensure all participants are accounted for and to troubleshoot message traffic?

**Comment by jrw972** [comment]
I'm not aware of any known issues that would prevent a Java/C++ combination.

There are occasionally bugs in the transport but it has been my experience that most of the time it is a configuration or environmental issue.  In this respect, there are a couple of ways you can go.

- You can install a DDS listener or an observer (https://objectcomputing.com/resources/publications/mnb/2021/06/09/logging-samples-json-opendds-apps) to see the participants, publications, and subscriptions in the domain.  If you see the right things and your application still doesn't work, then look at the qos to ensure that it is not causing a mismatch.  If you don't see the right things, then discovery isn't working and it's probably a configuration issue or something in the environment preventing discovery from working (like a firewall).
- The OpenDDS logs, specifically DCPSDebugLevel and TransportDebugLevel, can provide insight as to what is going on.  For example, if you are using RTPS discovery and you never see SPDP messages from a remote participant, then I would check the configuration on both sides to make sure that they are using the same multicast group, check that multicast is supported by your router, etc.
- Wireshark is useful for debugging configuration and connectivity issues.
- There is a new monitoring tool for RTPS users.  https://github.com/OpenDDS/opendds-monitor
- There is a monitoring tool for InfoRepo but it has not been maintained (and will go away is a future release).
